typesetters/base.lua

```diff
@@ -145,6 +145,13 @@ function typesetter.declareSettings(_)
     default = false,
     help = "When true, a warning is issued when a soft hyphen is encountered"
   })
+
+  SILE.settings:declare({
+    parameter = "typesetter.dialogue",
+    type = "boolean",
+    default = true,
```

Comment:
Just to limit possible unforeseen consequences, I'm thinking we should default this to `false`, then explicitly enable it to true in the French and Turkish language modules (and possible a more languages that may have taken any queues from French).

Reply:
As for unforeseen consequences, they'd _only_ occur for an emdash + spacing at the start of a paragraph. I don't see any good reason one would want a stretchable space there, even if the language usage don't rely on this for dialogue speaker changes. Rather the contrary, e.g. if we get a "loose" list input in English
```
— Step 1 stuff

— Step 2 more stuff
```

Could be controversial, but even in those case I'd prefer seeing a fixed space after the dash.
`

Reply:
You may have a point. I was thinking more with my programmer hat on and the wasted CPU cycles searching for character sequences in all input than I was the typography implications. It's quite possible the "unforeseen consequences" would actually be improvements in other edge cases. One might even argue that *any* line-leading punctuation should be followed by a fixed space...

I could argue this both ways, I haven't convinced myself.

Reply:
I beat on it a bit and came up with a 10 page stress test with 500k paragraphs beginning with and without possible matches. Over many runs I couldn't come up with more than a ±0.01 difference which is totally in the noise.

By the way for your reference the way I'm doing benchmarks is with the Nix flake because you can easily run any branch or commit without any local setup hastle at all. From any (non SILE source code) directory setup the file you want to run, then use `nix run` with your two targets and at one warmup run (which serves to build the Flake if you haven't happened to have run that exact commit before).

```console
$ hyperfine --warmup 1 'nix run github:sile-typesetter/sile/master -- bench.sil' 'nix run github:Omikhleia/sile/feat-emdash-dialogue -- bench.sil'
Benchmark 1: nix run github:sile-typesetter/sile/master -- bench.sil
  Time (mean ± σ):      4.414 s ±  0.046 s    [User: 4.250 s, System: 0.094 s]
  Range (min … max):    4.334 s …  4.481 s    10 runs

Benchmark 2: nix run github:Omikhleia/sile/feat-emdash-dialogue -- bench.sil
  Time (mean ± σ):      4.375 s ±  0.031 s    [User: 4.226 s, System: 0.079 s]
  Range (min … max):    4.336 s …  4.437 s    10 runs

Summary
  nix run github:Omikhleia/sile/feat-emdash-dialogue -- bench.sil ran
    1.01 ± 0.01 times faster than nix run github:sile-typesetter/sile/master -- bench.sil
```